unicore/hub/service/app_api.py

```diff
@@ -0,0 +1,107 @@
+from pyramid.exceptions import NotFound, Forbidden
+
+from cornice import Service
+from cornice.resource import resource, view
+
+from unicore.hub.service.models import App
+from unicore.hub.service.schema import App as AppSchema
+from unicore.hub.service import utils
+
+
+def require_principal(request, principals):
+    effective_principals = set(request.effective_principals)
+    accepted_principals = set(principals)
+    if not effective_principals.intersection(accepted_principals):
+        raise Forbidden
+
+
+def get_app_object(request):
+    app_id = request.matchdict['app_id']
+    app = request.db.query(App).get(app_id)
+
+    if app is None:
+        raise NotFound
+
+    return app
+
+
+@resource(collection_path='/apps', path='/apps/{app_id:\d+}')
```

Comment:
Part of it's a feeling I've got about separating credentials and display names (where there might be a need for readability) from lower level implementation. But perhaps the better reason is that the ID cannot change whereas a slug could change (not technically with my implementation, but it's not a database-level guarantee).

I'm happy to reconsider. It might help to know that a client library will abstract away the ID usage when invoking api endpoints.